ios/FluentUI/Vnext/HUD/HeadsUpDisplay.swift

```diff
@@ -0,0 +1,227 @@
+                        .frame(height: verticalPadding)
+                    Text(label)
+                        .foregroundColor(foregroundColor)
+                        .lineLimit(2)
+                        .multilineTextAlignment(.center)
+                        .fixedSize(horizontal: false, vertical: true)
+                }
+            }
+        }
+        .padding(EdgeInsets(top: verticalPadding,
+                            leading: horizontalPadding,
+                            bottom: verticalPadding,
+                            trailing: horizontalPadding))
+        .squareShaped(minSize: tokens.minSize,
+                      maxSize: tokens.maxSize)
+        .background(Rectangle()
+                        .fill(Color(dynamicColor: tokens.backgroundColor))
+                        .frame(maxWidth: .infinity,
+                               maxHeight: .infinity,
+                               alignment: .center)
+                        .cornerRadius(tokens.cornerRadius)
+        )
+        .contentShape(Rectangle())
+        .onChange(of: isPresented, perform: { present in
+            if present {
+                presentAnimated()
+            } else {
+                dismissAnimated()
+            }
+        })
+        .onAnimationComplete(for: presentationScaleFactor) {
+            resetScaleFactor()
+        }
+        .opacity(opacity)
+        .scaleEffect(presentationScaleFactor)
+        .onTapGesture {
+            state.tapAction?()
+        }
+    }
+
+    /// UIKit wrapper that exposes the SwiftUI Heads-up Display implementation.
+    /// - Parameters:
+    ///   - type: The `HUDType` enum value of the Heads-up Display.
+    ///   - isPresented: Binding boolean that controls whether the Heads-up Display is presented.
+    ///   - label: Label string presented by the Heads-up Display.
+    ///   - tapAction: Closure executed when the user taps on the Heads-up Display.
+    public init(type: HUDType,
+                isPresented: Binding<Bool>? = nil,
+                label: String? = nil,
+                tapAction: (() -> Void)? = nil) {
+        let stateImpl = MSFHUDStateImpl(type: type)
+        stateImpl.label = label
+        stateImpl.tapAction = tapAction
+        state = stateImpl
+
+        if let isPresented = isPresented {
+            _isPresented = isPresented
+            opacity = Constants.opacityDismissed
+            presentationScaleFactor = HUD.Constants.showAnimationScale
+        } else {
+            _isPresented = .constant(true)
+            opacity = Constants.opacityPresented
+            presentationScaleFactor = Constants.presentationScaleFactorDefault
+        }
+    }
+
+    let defaultTokens: HeadsUpDisplayTokens = .init()
+    var tokens: HeadsUpDisplayTokens {
+        return resolvedTokens
+    }
+    @Environment(\.fluentTheme) var fluentTheme: FluentTheme
+    @Binding var isPresented: Bool
+    @State var opacity: Double
+    @State var presentationScaleFactor: CGFloat
```

Comment:
Why are `opacity` and `presentationScaleFactor` stored as `@State` instead of on the `state` object?

Reply:
The rationale behind it is that they shouldn't be changed from anywhere else other than this View because they are all about the internal/intrinsic behavior of it.

I made them **private** to ensure this is not changed even within the library (as that would be the case if it was part of the state object).